### PR TITLE
Refactor old gens to for...of

### PR DESCRIPTION
**Pull request by whalemer**

### CODE REVIEW

File: mods/gen1/moves.js [hunk 1]
```diff
@@ -107,9 +107,9 @@ exports.BattleMovedex = {
 					return;
 				}
 				let moves = pokemon.moveset;
-				for (let i = 0; i < moves.length; i++) {
-					if (moves[i].id !== 'bide') {
-						pokemon.disableMove(moves[i].id);
+				for (let move of moves) {
```
Comment (Zarel) [inline mods/gen1/moves.js]: `const`

Comment (Zarel) [reply mods/gen1/moves.js]: Ditto every other instance of it not being modified.

Comment (Zarel) [reply mods/gen1/moves.js]: Also `for (const move of pokemon.moveset) {`

No need to cache `moves` when you don't need to access it repeatedly.

File: mods/gen1/moves.js [hunk 2]
```diff
@@ -429,13 +428,12 @@ exports.BattleMovedex = {
 						pokemon.setStatus('psn');
 					}
 					let volatiles = Object.keys(pokemon.volatiles);
-					for (let n = 0; n < volatiles.length; n++) {
-						let id = volatiles[n];
-						if (id === 'residualdmg') {
-							pokemon.volatiles[id].counter = 0;
+					for (let volatile of volatiles) {
```
Comment (Zarel) [inline mods/gen1/moves.js]: Ditto here. `for (const volatile of Object.keys(pokemon.volatiles)) {`

File: mods/gen1/random-teams.js [hunk 3]
```diff
@@ -332,7 +331,7 @@ class RandomGen1Teams extends RandomGen2Teams {
 						} // End of switch for moveid
 					}
 					if (rejected) {
-						moves.splice(k, 1);
+						moves.splice(moves.indexOf(moveid), 1);
```
Comment (Slayer95) [inline mods/gen1/random-teams.js]: Just keep the normal iteration imo

**Comment by Zarel** [comment]
Are you going to maintain this?

**Comment by Zarel** [comment]
Closing, unmaintained.

**Comment by KrisOnline** [comment]
I’ll do it when I get home if nobody else does by then

**Pull request closed without merging**